packages/relay/src/lib/eth.ts

```diff
@@ -2331,15 +2331,22 @@ export class EthImpl implements Eth {
           transactionIndex: nullableNumberTo0x(receiptResponse.transaction_index),
         });
       });
-
+      const functionSelectors = constants.HTS_CREATE_FUNCTIONS_SIGNATURE.map((signature) =>
+        Utils.calculateFunctionSelector(signature),
+      );
+      const isCreationViaSystemContract = functionSelectors.includes(
+        receiptResponse.function_parameters.substring(0, constants.FUNCTION_SELECTOR_CHAR_LENGTH),
+      );
```

Comment:
Тhis is not how regexes are supposed to be used, we are essentially doing the same thing with the regex, and it's arguable whether it's more efficient. The difference is going to be in nanoseconds, so I don't think it's worth the extra effort and loss of readability in the code.

If the array was bigger and we were really seeing a performance hit by this traversal, a simpler and more efficient optimization would be to use a `Set` instead of `Array` - checking if an element is contained within a set takes `O(1)` time. But we don't need to do that currently, with only 10 elements the difference would be negligible.

Reply:
I agree with Victor, looking for an element in an array is not usually done via regEx. Also, benchmarked the solution with includes vs regex for 1000 iterations:

![Screenshot 2024-12-06 at 15 45 21](https://github.com/user-attachments/assets/bff9a601-f8f1-4851-899c-35e506f4cc79)

Thats due to the facts that RegExp is not as optimized for such operations as includes. If you insist I may use a Set here, however its a small array and won't change much in time. I see it as additional and not necessary effort

Reply:
Nice thanks a lot guys, @victor-yanev and @konstantinabl! Really appreciate the information and the effort to confirm! You're right the array is small enough so the difference would be negligible! Let's stick to the original implementation if it's efficient enough!